lib/collection/src/shards/replica_set/update.rs

```diff
@@ -128,7 +129,7 @@ impl ShardReplicaSet {
 
         peer_ids
             .into_iter()
-            .filter(|peer_id| self.peer_is_active(peer_id)) // re-acquire replica_state read lock
+            .filter(|peer_id| self.peer_is_active_or_resharding(peer_id)) // re-acquire replica_state read lock
```

Comment:
Small fix: allows selecting replicas in `Resharding` state as a leader peer for `Medium` write ordering.

When resharding "up" we only have a single replica, and without this fix updates with `Medium` ordering would fail until all resharding transfers are finished and replica is switched into `Active`.